pkg/controller/trustbundle-publisher/controller.go

```diff
@@ -0,0 +1,206 @@
+// The trustbundle-publisher controller is responsible for merging a user
+// provided trust bundle with the system trust bundle and publishing the
+// merged bundle to a configmap in the "openshift-config-managed" namespace.
```

Comment:
ultimately this is going to be folded into the proxy controller right?  (so that we only perform this merge/publication after confirming we can pass proxy validation with these CAs)

Reply:
@bparees this controller will work with the proxy controller by watching/validating the user-provided add'l trust bundle configmap, merging the bundle with the system bundle and creating the combined bundle configmap that is consumed by the proxy controller. The proxy controller will use the combined bundle in the configmap to validate https readinessEndpoints and for https proxying by validating the bundle referenced in trustedCA of a proxy object.

Reply:
that still makes it sound like the combined bundle is published before it's been validated as working/usable.

meaning an admin could update the bundle to contain invalid content, which would be propagated to the downstream consumers and break proxy usage.

Reply:
@bparees https://github.com/openshift/cluster-network-operator/pull/271 validates the user provided trust bundle, merges with the system bundle to form the `trusted-ca-bundle` configmap. If a user-provided bundle does not exist, https://github.com/openshift/cluster-network-operator/pull/271/files#diff-6cc3b6cf53aa0e5d6919e9943b4b174aR151 creates a system bundle for the `trusted-ca-bundle` configmap to use when `proxy.spec.httpsProxy` is defined.